### PR TITLE
Add gRPC APIs for ContainerInfo and List:

### PR DESCRIPTION
**Pull request by feiskyer**
Add gRPC APIs for ContainerInfo and List (Part of #278):

* ContainerInfo
* PodList
* ContainerList
* ImageList
* VMList

### CODE REVIEW

File: serverrpc/list.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	glog.V(3).Infof("ContainerList with request %s", req.String())
+
+	containerList, err := s.daemon.List("container", req.PodID, req.VmID, req.Auxiliary)
+	if err != nil {
+		glog.Errorf("ContainerList error: %v", err)
+		return nil, err
+	}
+
+	if _, ok := containerList["cData"]; !ok {
+		return nil, nil
```
Comment (resouer) [inline serverrpc/list.go]: nil error, nil value. will this broke the caller?

Comment (feiskyer) [inline serverrpc/list.go]: Y, should be `&types.ContainerListResponse{ContainerList: nil}, nil`

File: serverrpc/images.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+func (s *ServerRPC) ImageList(ctx context.Context, req *types.ImageListRequest) (*types.ImageListResponse, error) {
+	glog.V(3).Infof("ImageList with request %s", req.String())
+
+	images, err := s.daemon.Daemon.Images(req.FilterArgs, req.Filter, req.All)
+	if err != nil {
+		glog.Errorf("ImageList error: %v", err)
+		return nil, err
+	}
+
+	result := make([]*types.ImageInfo, 0, 1)
```
Comment (laijs) [inline serverrpc/images.go]: make([]*types.ImageInfo, 0, len(images))

Comment (feiskyer) [inline serverrpc/images.go]: Updated.

File: serverrpc/list.go (hunk 1 above)
Comment (feiskyer) [inline serverrpc/list.go]: Updated.

**Pull request merged**